test/decl/typealias/generic.swift

```diff
@@ -1,9 +1,31 @@
-// RUN: %target-parse-verify-swift
+// RUN: %target-parse-verify-swift -enable-experimental-nested-generic-types
 
-struct MyType<TyA, TyB> {
+struct MyType<TyA, TyB> { // expected-note {{declared here}}
   var a : TyA, b : TyB
 }
 
+//
+// Type aliases that reference unbound generic types -- not really generic,
+// but they behave as such, in the sense that you can apply generic
+// arguments to them.
+//
+
+// FIXME: This should work?
+typealias OurType = MyType  // expected-error {{reference to generic type 'MyType' requires arguments in <...>}}
+
+typealias YourType = Swift.Optional
```

Comment:
Sorry, the brackets got swallowed, I meant: typealias YourType\<T\> = Swift.Optional\<T\>

Yes, I think it makes sense to be an error.  Lets start narrow, we can always expand the scope of the feature later if it makes sense (which isn't clear to me).